ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -209,9 +209,11 @@ func (rv *Clusters) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if rv.NumMasters != 1 && rv.NumMasters != 3 {
 		return fmt.Errorf("Number of masters (counted %d) must be exactly 1 or 3", rv.NumMasters)
 	}
-	// Autodetect ClusterType based on the node counts
+	// Autodetect ClusterType based on the node counts and fix number of workers to 0 for sno.
+	// The latter prevents AgentClusterInstall from being mutated upon SNO expansion
 	if rv.NumMasters == 1 {
 		rv.ClusterType = SNO
+		rv.NumWorkers = 0
```

Comment:
rv.NumWorkers is set to 0 at [L200](https://github.com/openshift-kni/cnf-features-deploy/pull/1208/files#diff-5a7eeb6977f4204a7d5c1322a7e1fe7957d13801d22c91348770ef8a71c22466R200)

Reply:
But then at line 206 it is incremented if a worker is added in siteconfig (SNO expansion case).
This exactly leads to my problem of AgentClusterInstall mutation

Reply:
okay, got it